### PR TITLE
Update to the latest Babylon Native and various build fixes

### PR DESCRIPTION
**Pull request by bghgary**
The latest Babylon Native has a number of build related changes that require updates on the Babylon React Native side.

Running Playground Locally:
- [x] Android
- [x] iOS
- [x] Windows

PackageTest:
- [ ] Android
- [x] iOS
- [ ] Windows

### CODE REVIEW

File: Modules/@babylonjs/react-native-windows/windows/CMakeLists.txt [hunk 1]
```diff
@@ -53,9 +40,3 @@ target_link_libraries(BabylonNative
     Window
     XMLHttpRequest
     Canvas)
-
-if(DEFINED EXTENSIONS_DIRS)
-    foreach(extensionDir IN LISTS EXTENSIONS_DIRS)
-        add_subdirectory(${extensionDir} "Extensions" EXCLUDE_FROM_ALL)
-    endforeach()
-endif()
```
Comment (ryantrem) [inline Modules/@babylonjs/react-native-windows/windows/CMakeLists.txt:61]: All this is no longer needed or something?

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp [hunk 2]
```diff
@@ -321,11 +323,13 @@ namespace Babylon
         return false;
     }
 
-    void UpdateXRView(void* windowPtr)
+#if defined(__APPLE__) || defined(ANDROID)
```
Comment (ryantrem) [inline Modules/@babylonjs/react-native/shared/BabylonNative.cpp:326]: Why is this needed? Don't we have a no-op impl on other platforms?

File: Package/gulpfile.js [hunk 3]
```diff
@@ -74,88 +74,78 @@ const initializeSubmodulesMostRecentBabylonNative = async () => {
...
-  shelljs.mkdir('-p', './../Modules/@babylonjs/react-native/Build/uwp_arm64');
-  exec('cmake -D CMAKE_SYSTEM_NAME=WindowsStore -D CMAKE_SYSTEM_VERSION=10.0 -D NAPI_JAVASCRIPT_ENGINE=JSI -A arm64 ./../../../react-native-windows/windows', './../Modules/@babylonjs/react-native/Build/uwp_arm64');
-}
+const makeUWPProjectx86 = async () => makeUWPProjectPlatform('x86', 'Win32');
+const makeUWPProjectx64 = async () => makeUWPProjectPlatform('x64', 'x64');
+const makeUWPProjectARM = async () => makeUWPProjectPlatform('arm', 'arm');
+const makeUWPProjectARM64 = async () => makeUWPProjectPlatform('arm64', 'arm64');
 
 const makeUWPProject = gulp.parallel(
   makeUWPProjectx86,
```
Comment (ryantrem) [inline Package/gulpfile.js:113]:
```suggestion
    makeUWPProjectx86,
```

**Review by ryantrem — APPROVED** [review]

File: Modules/@babylonjs/react-native-windows/windows/CMakeLists.txt (hunk 1 above)
Comment (bghgary) [reply Modules/@babylonjs/react-native-windows/windows/CMakeLists.txt:61]: We are now adding the top level CMakeLists, so it's already included in that.

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp (hunk 2 above)
Comment (bghgary) [reply Modules/@babylonjs/react-native/shared/BabylonNative.cpp:326]: Other platforms have a type that isn't convertible to `void*` which is what the underlying XR.h currently takes.

File: Package/gulpfile.js (hunk 3 above)
Comment (bghgary) [reply Package/gulpfile.js:113]: Woah, all the spacing changed. That wasn't intended. Will revert the spacing.

Comment (bghgary) [reply Package/gulpfile.js:107]: I just formatted the whole file using vscode.

**Comment by bghgary** [comment]
Will test packages on Android/Windows afterwards.
